Integration tests are flaky on CI

In my experience with recent PRs on the javascript SDK, I get failures like the following in CI in the integration tests:

```
Discover call with info ,8,akkaserverless-proxy-core,0.8.7,akkaserverless.component.action.Actions,akkaserverless.component.view.Views,akkaserverless.component.valueentity.ValueEntities,akkaserverless.component.replicatedentity.ReplicatedEntities,akkaserverless.component.eventsourcedentity.EventSourcedEntities,true,localhost, sending 1 components
    ✓ should increase non existing entity (910ms)
    ✓ should increase another entity (91ms)
    1) "after all" hook for "should increase another entity"


  2 passing (11s)
  1 failing

  1) Counter service
       "after all" hook for "should increase another entity":
     Uncaught Error: connect ECONNREFUSED 127.0.0.1:8080
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)



/home/circleci/project/samples/js/valueentity-counter/node_modules/mocha/lib/runner.js:962
    throw err;
```

The actual test and sample project that it fails on is different each time, but the failure is the same. I also noticed this error, but not every time:

```
Error: done() called multiple times in hook <Counter service "after all" hook for "should increase another entity"> of file /home/circleci/project/samples/js/valueentity-counter/integration-test/counter.test.js
    at createMultipleDoneError (/home/circleci/project/samples/js/valueentity-counter/node_modules/mocha/lib/errors.js:428:13)
    at multiple (/home/circleci/project/samples/js/valueentity-counter/node_modules/mocha/lib/runnable.js:290:24)
    at done (/home/circleci/project/samples/js/valueentity-counter/node_modules/mocha/lib/runnable.js:301:14)
    at /home/circleci/project/samples/js/valueentity-counter/node_modules/mocha/lib/runnable.js:414:7
    at wrappedCallback (/home/circleci/project/samples/js/valueentity-counter/node_modules/@grpc/grpc-js/build/src/server.js:490:13)
    at ServerHttp2Session.maybeCallback (/home/circleci/project/samples/js/valueentity-counter/node_modules/@grpc/grpc-js/build/src/server.js:496:17)
    at Object.onceWrapper (events.js:519:28)
    at ServerHttp2Session.emit (events.js:412:35)
    at emitClose (internal/http2/core.js:1052:8)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  code: 'ERR_MOCHA_MULTIPLE_DONE',
  valueType: 'undefined',
  value: undefined
}
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! valueentity-counter@0.0.1 integration-test: `mocha ./integration-test`
npm ERR! Exit status 7
npm ERR! 
npm ERR! Failed at the valueentity-counter@0.0.1 integration-test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I've looked into it briefly, nothing obvious stands out, the only thing that maybe is causing a problem is that we have some things that continually retry until it succeeds, and it's possible that these errors area actually being picked up by an unhandled exception catcher, so perhaps when it fails, these errors are being caught and causing the after all hook to fail. Not sure.

--- a/sdk/src/integration-testkit.ts
+++ b/sdk/src/integration-testkit.ts
@@ -116,14 +116,21 @@ export class IntegrationTestkit {
    * @param callback - shutdown callback, accepting possible error
    */
   shutdown(callback: (error?: any) => void) {
-    if (this.proxyContainer !== undefined) {
-      this.proxyContainer.stop();
-    }
-
     Object.getOwnPropertyNames(this.clients).forEach((client) => {
       this.clients[client].close();
     });
-
-    this.akkaServerless.tryShutdown(callback);
+    let proxyContainerStopped: Promise<void>;
+    if (this.proxyContainer !== undefined) {
+      // Important, ensure that the proxy container is stopped before we shut down
+      // ourselves, otherwise it will try to reconnect and that will cause unhandled
+      // exceptions.
+      proxyContainerStopped = this.proxyContainer.stop();
+    } else {
+      proxyContainerStopped = Promise.resolve();
+    }
+    proxyContainerStopped.then(
+      () => this.akkaServerless.tryShutdown(callback),
+      (err) => this.akkaServerless.tryShutdown(() => callback(err)),
+    );
   }
 }

